test/core/utils/preload-media.js

```diff
@@ -1,73 +1,84 @@
-describe('axe.utils.preloadMedia', function () {
-  'use strict';
+describe('axe.utils.preloadMedia', () => {
+  const fixtureSetup = axe.testUtils.fixtureSetup;
 
-  var fixture = document.getElementById('fixture');
-  var fixtureSetup = axe.testUtils.fixtureSetup;
-
-  afterEach(function () {
-    fixture.innerHTML = '';
-  });
-
-  it('returns empty array when there are no media nodes to be preloaded', function (done) {
+  it('returns empty array when there are no media nodes to be preloaded', async () => {
     axe._tree = axe.utils.getFlattenedTree(document);
 
-    axe.utils.preloadMedia({ treeRoot: axe._tree[0] }).then(function (result) {
-      assert.equal(result.length, 0);
-      done();
-    });
+    const result = await axe.utils.preloadMedia({ treeRoot: axe._tree[0] });
+    assert.equal(result.length, 0);
   });
 
-  it('returns empty array when <audio> has no source', function (done) {
+  it('returns empty array when <audio> has no source', async () => {
     fixtureSetup('<audio autoplay="true" controls></audio>');
 
-    axe.utils.preloadMedia({ treeRoot: axe._tree[0] }).then(function (result) {
-      assert.equal(result.length, 0);
-      done();
-    });
+    const result = await axe.utils.preloadMedia({ treeRoot: axe._tree[0] });
+    assert.equal(result.length, 0);
   });
 
-  it('returns empty array when <video> has no source', function (done) {
+  it('returns empty array when <video> has no source', async () => {
     fixtureSetup('<video id="target"><source src=""/></video>');
-    axe.utils.preloadMedia({ treeRoot: axe._tree[0] }).then(function (result) {
-      assert.equal(result.length, 0);
-      done();
-    });
+    const result = await axe.utils.preloadMedia({ treeRoot: axe._tree[0] });
+    assert.equal(result.length, 0);
   });
 
-  it('returns media node (audio) after their metadata has been preloaded', function (done) {
+  it('returns empty array when media node does not preload', async () => {
+    fixtureSetup(`
+      <video id="target" preload="none">
```

Comment:
Should be updated to set autoplay, otherwise the test would pass regardless of preload none.

(that's assuming the test browser we're using actually has the necessary autoplay behavior disabled; might need to compromise on this test if we can't work out a way to actually have the test browser skip preloading in that scenario)